packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/layout/useLayout.ts

```diff
@@ -109,9 +111,17 @@ export const useLayout = (): UseLayoutValue => {
     });
   }, []);
 
+  const arrangeAll = useCallback((afterLayoutCallback: (laidoutDiagram: RawDiagram) => void): void => {
```

Comment:
Why do you need `useCallback` here?

Reply:
I would have said that the arrangeAll function does not need to be created repeatedly over renders.

Reply:
It may also but used as useEffect dependency (like all functions in hook)

Reply:
- Creating the function over and over is not an issue per-se, we are creating tons of functions all the time everywhere
- Technically you are right, it would improve the performance of some code using it as a dependency in an `useEffect` but here, you are definitely NOT using the arrangeAll function in an `useEffect`

Some people, mostly those writing frameworks, are using `useMemo` and `useCallback` on everything they return to prevent issues in every possible use cases that their consumers will create. It has a memory impact and it makes the code more complex. I would say that this `useCallback` is not necessary of today so we don't need it because others will see that kind of code and try to put it everywhere.

The React Forget compiler is supposed to remove this need soon (in React 19 I think) so I don't want to put more `useMemo` / `useCallback` then necessary in our code base.

Reply:
I will remove the `useCallback`